.github/workflows/main.yml

```diff
@@ -36,10 +36,10 @@ jobs:
     - name: Checkout Code
       uses: actions/checkout@v4
 
-    - name: Set up JDK 11
+    - name: Set up JDK 17
```

Comment:
I prefer to build with minimum required version for build.

Why using JDK 21 ? :slightly_smiling_face: 

(Note that if this is about checking if built code works with java 21 with have : https://github.com/eclipse-leshan/leshan/blob/master/.jenkins/nightly.jenkins)

Reply:
Just to give more time before the next upgrade:  https://adoptium.net/support/
If it was a prod server I would have performance arguments :joy_cat:

Reply:
> performance arguments ? 

Which kind of gain we get ? You mean on build time OR this will produce better bytecode  OR something else ?

Reply:
since you asked: better scalability mainly (doesn't make sense in this context)
https://webtide.com/jetty-12-virtual-threads-support/
in my experience, Java 17 to Java 21 doesn't change much in terms of raw CPU/GC perf (a couple of percent max)

Reply:
Just to be sure this benefits are at execution time ? not build time ? 

I mean if I build Leshan with java 17 and execute server-demo with java 21, then Jetty will works faster thx to virutal thread, right ?  or I missed something ?